apps/extension/src/listeners/message-prax-request.ts

```diff
@@ -8,29 +8,37 @@ import { UserChoice } from '@penumbra-zone/types/user-choice';
 // this is the only message we handle from an unapproved content script.
 chrome.runtime.onMessage.addListener(
   (req: PraxConnection.Request | JsonValue, sender, respond: (arg: PraxConnection) => void) => {
-    if (req !== PraxConnection.Request) return false; // instruct chrome we will not respond
-
-    void approveOrigin(sender).then(
-      status => {
-        // user made a choice
-        if (status === UserChoice.Approved) {
-          respond(PraxConnection.Init);
-          void chrome.tabs.sendMessage(sender.tab!.id!, PraxConnection.Init);
-        } else {
-          respond(PraxConnection.Denied);
-        }
-      },
-      e => {
-        if (process.env['NODE_ENV'] === 'development') {
-          console.warn('Connection request listener failed:', e);
-        }
-        if (e instanceof ConnectError && e.code === Code.Unauthenticated) {
-          respond(PraxConnection.NeedsLogin);
-        } else {
-          respond(PraxConnection.Denied);
-        }
-      },
-    );
-    return true; // instruct chrome to wait for the response
+    console.log('prax request', req, sender);
+    switch (req) {
+      case PraxConnection.End:
+        void removeOrigin(sender);
+        respond(PraxConnection.End);
+        return true; // instruct chrome to wait for the response
+      case PraxConnection.Request:
```

Comment:
If `PraxConnection.Request` tries to get an approval, think we should rename this to `PraxConnection.RequestApproval`